libs/sensors/src/ProximitySwitch.cpp

```diff
@@ -0,0 +1,12 @@
+#include "ProximitySwitch.h"
+
+using namespace Sensor;
+
+ProximitySwitch::ProximitySwitch(PinName proximity_in) : m_proximity_in_adc(proximity_in) {}
+
+/*https://www.dfrobot.com/product-2025.html*/
+float ProximitySwitch::read() {
+  float proximity_reading = m_proximity_in_adc.read();
```

Comment:
Mbed has a concept of Not connected pin. So you will have to do a check that makes sure that the pin is connected. There is function `is_connected()` that you can use from the [DigitalIn API](https://os.mbed.com/docs/mbed-os/v6.15/mbed-os-api-doxy/classmbed_1_1_digital_in.html#acd4f2a88f26859acfb84737dbdd95345) for this purpose.